### PR TITLE
lib/ansible/parsing/ajson.py: added UNIT-tests

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
unit tests for lib/ansible/parsing/ajson.py AnsibleJSONEncoder

If it is OK, I'm going to add tests for other types when this PR will be merged

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by Andersson007** [comment]
+label: bugfix

File: test/units/parsing/test_ajson.py [hunk 1]
```diff
@@ -20,3 +24,66 @@ def test_AnsibleJSONDecoder_vault():
     assert isinstance(data['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['bar']['baz'][0]['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['foo']['password'], AnsibleVaultEncryptedUnicode)
+
+
+class TestAnsibleJSONEncoder():
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]: No need for a class.

File: test/units/parsing/test_ajson.py [hunk 2]
```diff
@@ -20,3 +24,66 @@ def test_AnsibleJSONDecoder_vault():
...
+
+            def __iter__(self):
+                return iter(self.__dict__)
+
+            def __len__(self):
+                return len(self.__dict__)
+
+        yield M(request.param)
+
+    @pytest.yield_fixture
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]:
```suggestion
    @pytest.fixture
```

File: test/units/parsing/test_ajson.py [hunk 3]
```diff
@@ -20,3 +24,66 @@ def test_AnsibleJSONDecoder_vault():
...
     assert isinstance(data['bar']['baz'][0]['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['foo']['password'], AnsibleVaultEncryptedUnicode)
+
+
+class TestAnsibleJSONEncoder():
+    """
+    Class for testing AnsibleJSONEncoder
+    """
+
+    @pytest.yield_fixture
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]:
```suggestion
    @pytest.fixture
```

File: test/units/parsing/test_ajson.py (hunk 1 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: @mattclay , yes. I’d like to organize tests for each class of ajson.py by particular classes into the test file.
If new classes were added to ajson.py, I think, it will be more comfortable to work with test in this way

File: test/units/parsing/test_ajson.py (hunk 2 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: thank you for clarifying about deprecated decorators.
changed

**Comment by Andersson007** [comment]
@mattclay , AnsibleJSONEncoder is completely covered. PTAL
ready_for_review

File: test/units/parsing/test_ajson.py [hunk 4]
```diff
@@ -20,3 +25,159 @@ def test_AnsibleJSONDecoder_vault():
...
+    def test_mapping(self, ansible_json_encoder, mapping, expected):
+        """
+        Test for passing Mapping object to AnsibleJSONEncoder.default()
+        """
+        m = mapping
+        assert(isinstance(m, Mapping))
+        assert(ansible_json_encoder.default(m) == expected)
+
+    # The pytest.mark.parametrize decorator is not used for this function
+    # because, at least, it's difficult to read it in this particular case.
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]: If you're concerned about the readability of having the data inline in the parametrize decorator, why not put it in a variable and reference that from the decorator instead?

File: test/units/parsing/test_ajson.py [hunk 5]
```diff
@@ -20,3 +25,159 @@ def test_AnsibleJSONDecoder_vault():
...
+        # previously check that test data is a suitable type:
+        assert(isinstance(data_0, AnsibleVaultEncryptedUnicode))
+        assert(isinstance(data_1, AnsibleVaultEncryptedUnicode))
+
+        # main assertions:
+        assert(ansible_json_encoder.default(data_0) == {'__ansible_vault': expected_0})
+        assert(ansible_json_encoder.default(data_1) == {'__ansible_vault': expected_1})
+
+    @pytest.mark.parametrize('test_input,expected', [(unsafe(), ''), ])
+    def test_ansible_unsafe(self, ansible_json_encoder, test_input, expected):
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]: If this test won't be implemented it should be removed.

File: test/units/parsing/test_ajson.py [hunk 6]
```diff
@@ -20,3 +25,159 @@ def test_AnsibleJSONDecoder_vault():
...
+            (set('set'), ''),
+        ]
+    )
+    def test_default_encoder_unserializable(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing unserializable objects of different classes.
+        It must fail with TypeError 'object is not serializable'
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py]: Check for the exception in the test body instead of using a decorator. Also, since you're testing for an exception, you only need to define the input and not compare against expected:

```suggestion
        with pytest.raises(TypeError):
            ansible_json_encoder.default(test_input)
```

File: test/units/parsing/test_ajson.py [hunk 7]
```diff
@@ -20,3 +25,159 @@ def test_AnsibleJSONDecoder_vault():
...
+class TestAnsibleJSONEncoder():
+    """
+    Class for testing AnsibleJSONEncoder
+    """
+
+    #################
+    # Fixture section
+
+    @pytest.fixture
+    def mapping(self, request):
```
Comment (mattclay) [inline test/units/parsing/test_ajson.py:72]: Include a comment about why you're using this class instead of a plain dict.

I assume you chose this approach so that the interface exposed is Mapping instead of MutableMapping.

**Comment by mattclay** [comment]
@Andersson007 Thanks for working on these tests. They're looking pretty good. Have you tried running the tests with [code coverage](https://docs.ansible.com/ansible/latest/dev_guide/testing_units.html#code-coverage-for-new-or-updated-unit-tests) to see if there are any gaps you can fill?

**Comment by Andersson007** [comment]
@mattclay, thank you for reviewing! Answering your question, no, I haven’t.
I’ve never heard about it. Thank you for the link!
I’ll try to figure it out and fix all points tomorrow because unfortunately evening in my location:), then ping you

**Comment by Andersson007** [comment]
Very good experience!

File: test/units/parsing/test_ajson.py (hunk 5 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: agreed, good point! fixed

File: test/units/parsing/test_ajson.py (hunk 6 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: agreed, it is more obvious. fixed

File: test/units/parsing/test_ajson.py (hunk 7 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py:72]: Done, thank you

File: test/units/parsing/test_ajson.py (hunk 4 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: @mattclay , I moved it to auxiliary function, PTAL

File: test/units/parsing/test_ajson.py [hunk 8]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+
+    expected_1 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3338\n')
+
+    return dict(
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: Please use a dict literal instead of a constructor.
```suggestion
    return {
```

File: test/units/parsing/test_ajson.py [hunk 9]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
     assert isinstance(data['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['bar']['baz'][0]['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['foo']['password'], AnsibleVaultEncryptedUnicode)
+
+
+def vault_data():
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py:30]: This function must be a fixture. You invoke it multiple times in tests which causes rereading the same file. Which in turn slows down tests execution.
```suggestion
@pytest.fixture
def vault_data():
```

File: test/units/parsing/test_ajson.py (hunk 1 above)
Comment (webknjaz) [reply test/units/parsing/test_ajson.py]: @Andersson007 it's [a premature optimization](https://stackify.com/premature-optimization-evil/). Currently, the class is unnecessary.

File: test/units/parsing/test_ajson.py [hunk 10]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+
+    return dict(
+        data_0=data_0,
+        data_1=data_1,
+        expected_0=expected_0,
+        expected_1=expected_1,
+    )
+
+
+class TestAnsibleJSONEncoder():
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: Also, in Python, you have to specify a list of base classes when declaring a new one or omit the braces completely.
```suggestion
class TestAnsibleJSONEncoder:
```

File: test/units/parsing/test_ajson.py [hunk 11]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+    """
+    Class for testing AnsibleJSONEncoder
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: That's not true. This class is used only to namespace things. Not for testing.

Also, [stop writing classes](https://www.youtube.com/watch?v=o9pEzgHorH0). It's not worth it, not in this case.

File: test/units/parsing/test_ajson.py [hunk 12]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+            (date(2019, 5, 14), '2019-05-14'),
+            (date(2020, 5, 14), '2020-05-14'),
+        ]
+    )
+    def test_date_datetime(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for passing datetime.date or datetime.datetime objects
+        to AnsibleJSONEncoder.default()
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: It's a statement, not a function.
```suggestion
        assert ansible_json_encoder.default(test_input) == expected
```

File: test/units/parsing/test_ajson.py [hunk 13]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+            ({1: 2}, {1: 2}),
+            ({2: 1}, {2: 1}),
+        ], indirect=['mapping']
+    )
+    def test_mapping(self, ansible_json_encoder, mapping, expected):
+        """
+        Test for passing Mapping object to AnsibleJSONEncoder.default()
+        """
+        m = mapping
+        assert(isinstance(m, Mapping))
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: This assertion doesn't make sense: you're testing whether your fixture returns a Mapping instance which it always does.
```suggestion
```

File: test/units/parsing/test_ajson.py [hunk 14]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+    @pytest.fixture
+    def mapping(self, request):
+        """
+        Returns object of Mapping mock class.
+        The object is used for testing handling of Mapping objects
+        in AnsibleJSONEncoder.default().
+        Using a plain dictionary instead is not suitable because
+        it is handled by default encoder of the superclass (json.JSONEncoder).
+        """
+        class M(Mapping):
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py:82]: This is evaluated on each fixture invocation, move it to the outer namespace.

File: test/units/parsing/test_ajson.py [hunk 15]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+    def ansible_json_encoder(self):
+        return AnsibleJSONEncoder()
+
+    ###############
+    # Test methods:
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (datetime(2019, 5, 14, 13, 39, 38, 569047), '2019-05-14T13:39:38.569047'),
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py:111]: Could you also add tests with timezones attached to datetimes

File: test/units/parsing/test_ajson.py [hunk 16]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+        assert(ansible_json_encoder.default(test_input) == expected)
+
+    @pytest.mark.parametrize(
+        'mapping,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ({2: 2}, {2: 2}),
+            ({1: 2}, {1: 2}),
+            ({2: 1}, {2: 1}),
+        ], indirect=['mapping']
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: I think it's nicer to have trailing commas:
```suggestion
        ], indirect=['mapping'],
```

File: test/units/parsing/test_ajson.py [hunk 17]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+    )
+    def test_mapping(self, ansible_json_encoder, mapping, expected):
+        """
+        Test for passing Mapping object to AnsibleJSONEncoder.default()
+        """
+        m = mapping
+        assert(isinstance(m, Mapping))
+        assert(ansible_json_encoder.default(m) == expected)
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]:
```suggestion
        assert ansible_json_encoder.default(m) == expected
```

File: test/units/parsing/test_ajson.py [hunk 18]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+        [
+            (vault_data()['data_0'], vault_data()['expected_0']),
+            (vault_data()['data_1'], vault_data()['expected_1']),
+        ]
+    )
+    def test_ansible_json_decoder_vault(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for passing AnsibleVaultEncryptedUnicode to AnsibleJSONEncoder.default()
+        """
+        assert(isinstance(test_input, AnsibleVaultEncryptedUnicode))
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: Same as above: unnecessary fixture test.

File: test/units/parsing/test_ajson.py [hunk 19]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+        assert(ansible_json_encoder.default(test_input) == {'__ansible_vault': expected})
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]:
```suggestion
        assert ansible_json_encoder.default(test_input) == {'__ansible_vault': expected}
```

File: test/units/parsing/test_ajson.py [hunk 20]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+            ({1: 'first'}, {1: 'first'}),
+            ({2: 'second'}, {2: 'second'}),
+        ]
+    )
+    def test_default_encoder(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing objects of different classes that are not tested above
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]:
```suggestion
        assert ansible_json_encoder.default(test_input) == expected
```

File: test/units/parsing/test_ajson.py [hunk 21]
```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
...
+
+    @pytest.mark.parametrize('test_input', [1, 1.1, 'string', [1, 2], set('set'), True, None])
+    def test_default_encoder_unserializable(self, ansible_json_encoder, test_input):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing unserializable objects of different classes.
+        It must fail with TypeError 'object is not serializable'
+        """
+        with pytest.raises(TypeError):
+            assert(ansible_json_encoder.default(test_input))
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: If you expect this invocation to emit an exception, `assert`ion doesn't make sense here:
```suggestion
            ansible_json_encoder.default(test_input)
```

**Review by webknjaz — DISMISSED** [review]
Thanks for the effort, yet I've spotted a few problems with the patch. Could you please fix those?

File: test/units/parsing/fixtures/ajson.json [hunk 22]
```diff
@@ -6,7 +6,7 @@
         "baz": [
             {
                 "password": {
-                    "__ansible_vault": "$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316562356435376162633631326264383934326565333633366238\n3863373264326461623132613931346165636465346337310a326434313830316337393263616439\n64653937313463396366633861363266633465663730303633323534363331316164623237363831\n3536333561393238370a313330316263373938326162386433313336613532653538376662306435\n3339\n"
+                    "__ansible_vault": "$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316562356435376162633631326264383934326565333633366238\n3863373264326461623132613931346165636465346337310a326434313830316337393263616439\n64653937313463396366633861363266633465663730303633323534363331316164623237363831\n3536333561393238370a313330316263373938326162386433313336613532653538376662306435\n3338\n"
```
Comment (webknjaz) [inline test/units/parsing/fixtures/ajson.json:9]: Why did you change this?

Comment (Andersson007) [reply test/units/parsing/fixtures/ajson.json:9]: because all instances have the same data. I think it does not make sense to pass the same to the test multiple times

Comment (webknjaz) [reply test/units/parsing/fixtures/ajson.json:9]: Instances of what?

File: test/units/parsing/test_ajson.py (hunk 10 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: @webknjaz I know that this class is unnecessary, it's a way to organize code

Comment (webknjaz) [reply test/units/parsing/test_ajson.py]: I've explained why you don't need this in other comments.

File: test/units/parsing/test_ajson.py (hunk 9 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py:30]: I tried to pass it as a fixture but failed. If you know a working way to pass it in the decorator as a fixture, please, share it

File: test/units/parsing/test_ajson.py (hunk 10 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: a separate class for each tested function/class-method looks better for me then separation by comments

File: test/units/parsing/test_ajson.py (hunk 11 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: 1. Good note about namespace.
2. Why should I stop writing classes? I do love them :)

File: test/units/parsing/test_ajson.py (hunk 12 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: thank you, I figured it out, good note!

File: test/units/parsing/test_ajson.py (hunk 9 above)
Comment (webknjaz) [reply test/units/parsing/test_ajson.py:30]: In the decorator? What do you mean?

File: test/units/parsing/test_ajson.py (hunk 10 above)
Comment (webknjaz) [reply test/units/parsing/test_ajson.py]: I didn't say anything about separation by comments. You don't need any separation when there's nothing to separate.

File: test/units/parsing/test_ajson.py (hunk 11 above)
Comment (webknjaz) [reply test/units/parsing/test_ajson.py]: you can use modules for namespacing. using classes for this is overkill

Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: I change the class comment, thank you

File: test/units/parsing/test_ajson.py (hunk 13 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: agreed

File: test/units/parsing/test_ajson.py (hunk 14 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py:82]: scope='class'

File: test/units/parsing/test_ajson.py (hunk 16 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: yep, done

File: test/units/parsing/test_ajson.py (hunk 18 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: removed

File: test/units/parsing/test_ajson.py (hunk 14 above)
Comment (webknjaz) [reply test/units/parsing/test_ajson.py:82]: no, it's in the method. it shouldn't be re-declared on every method invocation.

File: test/units/parsing/test_ajson.py [hunk 23]
```diff
@@ -20,3 +24,144 @@ def test_AnsibleJSONDecoder_vault():
...
+
+    expected_1 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3338\n')
+
+    return {
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: Return
```suggestion
    return [
        (data_0, expected_0),
        (data_1, expected_1),
    ]
```

And then have just one invocation.

File: test/units/parsing/test_ajson.py [hunk 24]
```diff
@@ -20,3 +24,144 @@ def test_AnsibleJSONDecoder_vault():
...
+    )
+    def test_mapping(self, ansible_json_encoder, mapping, expected):
+        """
+        Test for passing Mapping object to AnsibleJSONEncoder.default()
+        """
+        assert ansible_json_encoder.default(mapping) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]:
```suggestion
        vault_data(),
```

File: test/units/parsing/fixtures/ajson.json (hunk 22 above)
Comment (Andersson007) [reply test/units/parsing/fixtures/ajson.json:9]: in test/units/parsing/fixtures/ajson.json, all __ansible_vault values are the same

File: test/units/parsing/test_ajson.py (hunk 24 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py]: this is a hint! :) than you, done

File: test/units/parsing/test_ajson.py (hunk 9 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py:30]: seems we fixed it by another way

File: test/units/parsing/test_ajson.py (hunk 15 above)
Comment (Andersson007) [reply test/units/parsing/test_ajson.py:111]: done, thank you

**Comment by Andersson007** [comment]
@webknjaz , it's good that you've come :) Best experience for me. PTAL

File: test/units/parsing/test_ajson.py [hunk 25]
```diff
@@ -20,3 +25,138 @@ def test_AnsibleJSONDecoder_vault():
...
+    # Test methods:
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (datetime(2019, 5, 14, 13, 39, 38, 569047), '2019-05-14T13:39:38.569047'),
+            (datetime(2019, 5, 14, 13, 47, 16, 923866), '2019-05-14T13:47:16.923866'),
+            (date(2019, 5, 14), '2019-05-14'),
+            (date(2020, 5, 14), '2020-05-14'),
+            (datetime(2019, 6, 15, 14, 45, tzinfo=tz("UTC")), '2019-06-15T14:45:00+00:00'),
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]: Could you please keep the same string quoting style across the module?
```suggestion
            (datetime(2019, 6, 15, 14, 45, tzinfo=tz('UTC')), '2019-06-15T14:45:00+00:00'),
```

File: test/units/parsing/test_ajson.py [hunk 26]
```diff
@@ -20,3 +25,138 @@ def test_AnsibleJSONDecoder_vault():
...
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (datetime(2019, 5, 14, 13, 39, 38, 569047), '2019-05-14T13:39:38.569047'),
+            (datetime(2019, 5, 14, 13, 47, 16, 923866), '2019-05-14T13:47:16.923866'),
+            (date(2019, 5, 14), '2019-05-14'),
+            (date(2020, 5, 14), '2020-05-14'),
+            (datetime(2019, 6, 15, 14, 45, tzinfo=tz('UTC')), '2019-06-15T14:45:00+00:00'),
+            (datetime(2019, 6, 15, 14, 45, tzinfo=tz("Europe/Helsinki")), '2019-06-15T14:45:00+01:40'),
```
Comment (webknjaz) [inline test/units/parsing/test_ajson.py]:
```suggestion
            (datetime(2019, 6, 15, 14, 45, tzinfo=tz('Europe/Helsinki')), '2019-06-15T14:45:00+01:40'),
```

**Review by webknjaz — APPROVED** [review]
LGTM

**Comment by webknjaz** [comment]
@mattclay feel free to merge if you don't have any extra comments.

**Review by mattclay — APPROVED** [review]

**Comment by Andersson007** [comment]
@webknjaz , @mattclay , thank you for reviewing!
Feel free to ping me if you need some reviews too, I’ll be glad to help
